### ISSUE TITLE
Numbers are being mangled during posn creation/access

### ISSUE THREAD

**Issue by schanzer** [ds26gte/code.pyret.org#18]
Okay, this one took a LONG time to reduce....

```
(define-struct a-posn (x y))
(define a (* (a-posn-y (make-a-posn 0 -0.01)) 0.1))
(define b (* (  posn-y (  make-posn 0 -0.01)) 0.1))
```

In WeScheme, evaluating `b` produces `-0.001`, but Spyret throws an exception from within `toRepeatingDecimal()`.

Spyret has no problems with `a`, so I suspect something is going on with type information when it comes to Posns.

**Comment by ds26gte** [comment]
Fixed in ds26gte/pyret-lang@83f32fe.

When ``wrap``ing JS doubles, ensure that non-integers get converted to boxed rationals.

**Comment by schanzer** [comment]
Confirmed!
